### DISCUSSION TITLE
Что можно сделать?

### DISCUSSION THREAD

**Discussion by nikita946** [vvip-68/GyverPanelWiFi#311] [Задавайте вопросы] [answered]
Подскажите что можно сделать? Все делаю как по инструкции, но после прошивки на ESP8266, просто мигает светодиод, а на порте пусто. Еще при прошивке постоянно ругается как то вот так:
 c:\Users\admin\OneDrive\���������\Arduino\libraries\FastLED\src/fastspi.h:157:23: note: '#pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output'
  157 | #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from c:\Users\admin\OneDrive\���������\Arduino\libraries\FastLED\src\FastLED.h:79,
                 from c:\Users\admin\OneDrive\���������\Arduino\libraries\FastLED\src\five_bit_hd_gamma.cpp:1:
c:\Users\admin\OneDrive\���������\Arduino\libraries\FastLED\src\fastspi.h:157:23: note: '#pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output'
  157 | #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output".
До этого вообще не прошивалось, была ошибка компиляции, после обновления программы arduino, ошибка ушла.

**Comment by vvip-68** [comment]
Здесь нет никакой ошибки. Но если вас раздражают предупреждения от фастлед, что используется не аппаратный, а программный SPI можно
1. Купить ленту с другими светодиодами, у которых есть аппаратная шина SPI
2. Залезть в исходники FastLED, найти где выводится это предупреждения и закомментировать его.

**Reply by nikita946** [reply]
Хорошо. Почему тогда после завершения прошивки светодиод на ESP просто непрерывно мигает?

**Reply by vvip-68** [reply]
Варианты:
1. Что-то сделали не так.
2. Выбран пин вывода, связанный со встроенным светодиодом, поэтому когда идет вывод на этот пин - светодиод мигает.

А что в это время в мониторе порта?

**Reply by nikita946** [reply]
Просто какие то символы, и ничего внятного

**Reply by vvip-68** [reply]
Скорость порта согласована со скоростью, установленной в скетче?
В скетче по умолчанию 115200

**Reply by nikita946** [reply]
Сейчас вот опять прошил все заработало, но не могу зайти в веб интерфейс. Точка доступа создана, подключаюсь, далее по IP. В браузере пишет что не может получить доступ к сайту

**Reply by vvip-68** [reply]
Эта дискуссия - в ветке версии 1.13
В версии 1.13 нет никакого интерфейса в браузере - управляется только через приложение.
Залейте версию 1.14 - там есть веб-интерфейс. Не забудьте загрузить файловую систему микроконтроллера в которой находятся файлы скриптов веб-интерфейса.

**Reply by nikita946** [reply]
Скачал. Начиная прошивать и выходит вот так вот
Compilation error: grpc: error while marshaling: string field contains invalid UTF-8

**Reply by vvip-68** [reply]
Вот тут ничего не подскажу.  Вы в скрипте ничего не правили?
Ругается на какой-то кириллический символ там, где его быть не должно. Или файл битый.
Подробности есть?  Номер строки там, имя файла

**Reply by nikita946** [reply]
c:\Users\admin\OneDrive\���������\Arduino\libraries\FastLED\src/third_party/arduinojson/json.hpp:8127:2: error: #error ARDUINOJSON_SLOT_OFFSET_SIZE has been removed, use ARDUINOJSON_SLOT_ID_SIZE instead
 8127 | #error ARDUINOJSON_SLOT_OFFSET_SIZE has been removed, use ARDUINOJSON_SLOT_ID_SIZE instead
      |  ^~~~~
grpc: error while marshaling: string field contains invalid UTF-8

Compilation error: grpc: error while marshaling: string field contains invalid UTF-8

**Reply by vvip-68** [reply]
FasdtLED стоит какой-то случайной версии, вместо требуемой проекту 3.6.0

**Reply by nikita946** [reply]
Установил версию требуемую проектом. Все ровно ругается на тоже самое. Пробовал скачать заново проект, и опять та же самая ошибка.

**Reply by nikita946** [reply]
In file included from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_hard.h:630,
                 from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\timeProcessor.cpp:23:
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:39:6: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:39:14: warning: multi-character character constant [-Wmultichar]
In file included from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_hard.h:630,
                 from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\timeProcessor.cpp:23:
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:343:6: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:343:14: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:638:6: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:638:14: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:933:6: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:933:14: warning: multi-character character constant [-Wmultichar]
In file included from c:\Users\admin\OneDrive\���������\Arduino\libraries\FastLED\src/FastLED.h:75,
                 from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_hard.h:656,
                 from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\timeProcessor.cpp:23:
c:\Users\admin\OneDrive\���������\Arduino\libraries\FastLED\src/fastspi.h:157:23: note: '#pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output'
  157 | #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_hard.h:630,
                 from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:290:
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:39:6: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:39:14: warning: multi-character character constant [-Wmultichar]
In file included from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_hard.h:630,
                 from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:290:
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:343:6: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:343:14: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:638:6: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:638:14: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:933:6: warning: multi-character character constant [-Wmultichar]
C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_lang.h:933:14: warning: multi-character character constant [-Wmultichar]
In file included from c:\Users\admin\OneDrive\���������\Arduino\libraries\FastLED\src/FastLED.h:75,
                 from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\a_def_hard.h:656,
                 from C:\Users\admin\OneDrive\Рабочий стол\LedPanelWiFi-main\firmware\LedPanelWiFi_v1.14\LedPanelWiFi_v1.14.ino:290:
c:\Users\admin\OneDrive\���������\Arduino\libraries\FastLED\src/fastspi.h:157:23: note: '#pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output'
  157 | #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
. Variables and constants in RAM (global, static), used 44148 / 80192 bytes (55%)
║   SEGMENT  BYTES    DESCRIPTION
╠══ DATA     1848     initialized variables
╠══ RODATA   9428     constants       
╚══ BSS      32872    zeroed variables
. Instruction RAM (IRAM_ATTR, ICACHE_RAM_ATTR), used 61711 / 65536 bytes (94%)
║   SEGMENT  BYTES    DESCRIPTION
╠══ ICACHE   32768    reserved space for flash instruction cache
╚══ IRAM     28943    code in IRAM    
. Code in flash (default, ICACHE_FLASH_ATTR), used 685508 / 1048576 bytes (65%)
║   SEGMENT  BYTES    DESCRIPTION
╚══ IROM     685508   code in flash   
grpc: error while marshaling: string field contains invalid UTF-8

Compilation error: grpc: error while marshaling: string field contains invalid UTF-8

**Reply by vvip-68** [reply]
Ну, тогда не знаю. Не сталкивался с таким.
Попробуйте в настройках среды в отмеченном месте поставить "По умолчанию", если стоит что-то другое...

![image](https://github.com/user-attachments/assets/f5d58443-17fa-4bed-8548-23970dfe584e)

**Comment by nikita946** [comment]
В общем все ровно все тоже самое. Уже все испробовал

**Reply by vvip-68** [reply]
Ну, поставьте классическую Arduino IDE 1.8.19

**Reply by vvip-68** [reply, marked as answer]
Вот тут пишут, что это глюк 2.3.3 и советуют откатиться на 2.3.2
https://forum.arduino.cc/t/compilation-error-grpc-error-while-marshaling-string-field-contains-invalid-utf-8/1305471/5

Заметьте, что у всех жалующихся в этом посте - в пути к файлам, как и у вас - не английские буквы.
Может с этим как-то связано

**Reply by nikita946** [reply]
Спасибо за помощь)
